sdk/program/src/alt_bn128.rs

```diff
@@ -170,15 +176,22 @@ mod target_arch {
         )
         .try_into()?;
 
-        let mut result_point_data = [0; ALT_BN128_ADDITION_OUTPUT_LEN + 1];
         let result_point = p + q;
-        <G1 as ToBytes>::write(&result_point, &mut result_point_data[..])
+
+        let mut result_point_data = [0u8; ALT_BN128_ADDITION_OUTPUT_LEN];
+        let result_point_affine: G1 = result_point
+            .try_into()
+            .map_err(|_| AltBn128Error::ProjectiveToG1Failed)?;
+        result_point_affine
+            .x
+            .serialize_with_mode(&mut result_point_data[..32], Compress::No)
+            .map_err(|_| AltBn128Error::InvalidInputData)?;
+        result_point_affine
+            .y
+            .serialize_with_mode(&mut result_point_data[32..], Compress::No)
             .map_err(|_| AltBn128Error::InvalidInputData)?;
 
-        if result_point == G1::zero() {
```

Comment:
The zero special case was removed, does this still hold with the new code above?

Reply:
ark-works's serialization changed, with version 0.3.0 G1::zero did not serialize to [0u8;64], with 0.4.0 and serializing the x and y coordinates separately it does serialize to [0u8;64]. I have added a test for this.